extensions/builder.md

```diff
@@ -0,0 +1,180 @@
+    - [`io.buildpacks.builder.metadata` (JSON)](#iobuildpacksbuildermetadata-json)
+    - [`io.buildpacks.buildpack.order` (JSON)](#iobuildpacksbuildpackorder-json)
+    - [`io.buildpacks.buildpack.layers` (JSON)](#iobuildpacksbuildpacklayers-json)
+    - [`io.buildpacks.lifecycle.apis` (JSON)](#iobuildpackslifecycleapis-json)
+
+## General Requirements
+A **builder** is an image that MUST contain an implementation of the lifecycle, and build-time environment, and MAY contain buildpacks. Platforms SHOULD use builders to ease the build process.
+
+### Builder API Version
+This document specifies Builder API version `0.1`.
+
+Builder API versions:
+- MUST be in form `<major>.<minor>` or `<major>`, where `<major>` is equivalent to `<major>.0`
+- When `<major>` is greater than `0` increments to `<minor>` SHALL exclusively indicate additive changes
+
+### File/Directories
+A builder MUST have the following directories/files:
+- `/cnb/order.toml` &rarr; As defined in the [platform specification][order-toml-spec]
+- `/cnb/stack.toml` &rarr; As defined in the [platform specification][stack-toml-spec]
+- `/cnb/lifecycle/<lifecycle binaries>` &rarr; An implementation of the lifecycle, which contains the required lifecycle binaries for [building images][lifecycle-for-build].
+
+In addition, every buildpack blob contained on a builder MUST be stored at the following file path:
+- `<CNB_BUILDPACKS_DIR>/<buildpack ID>/<buildpack version>/`
+
+If the buildpack ID contains a `/`, it MUST be replaced with `_` in the directory name.
+
+All specified files and directories are writeable by the build environment's User. 
+
+### Environment Variables
+A builder MUST be an extension of a build-image, and MUST retain all specified environment variables and labels set on the original build image, as specified in the [platform specifications][build-image-specs].
+
+The following environment variables MUST be set on the builder:
+
+| Env Variable       | Description                                                                                         |
+| ------------------ | --------------------------------------------------------------------------------------------------- |
+| `CNB_APP_DIR`      | Application directory of the build environment (eg: `/workspace`)                                   |
+| `CNB_LAYERS_DIR`   | The directory to create and store `layers` in the build environment (eg: `/layers`)                 |
+| `CNB_PLATFORM_DIR` | The directory to create and store platform focused files in the build environment (eg: `/platform`) |
+
+The following variables MAY be set in the builder environment (through the image config's `Env` field):
+
+| Env Variable           | Description                            | Default |
+| ---------------------- | -------------------------------------- | ---- |
+| `SERVICE_BINDING_ROOT` | The directory where services are bound | - |
+| `CNB_BUILDPACKS_DIR` | The directory where buildpacks are located | `/cnb/buildpacks` |
```

Comment:
Why is this one optional when `CNB_LAYERS_DIR` etc. are not?

Reply:
I thought that it's a sensible default. I can make it required if you'd like.

Reply:
Since the lifecycle will supply default values for all of these variables if they are not set, I think it's reasonable to list defaults for all of them. Made a suggestion to that effect.